BrewMesModulesParent/machine/src/main/java/com/brewmes/machine/MachineServiceImpl.java

```diff
@@ -0,0 +1,256 @@
+package com.brewmes.machine;
+
+import com.brewmes.common.entities.Connection;
+import com.brewmes.common.services.IMachineService;
+import com.brewmes.common.util.Command;
+import com.brewmes.common.util.Products;
+import com.brewmes.common.util.machinenodes.CommandNodes;
+import com.brewmes.common_repository.ConnectionRepository;
+import org.eclipse.milo.opcua.sdk.client.OpcUaClient;
+import org.eclipse.milo.opcua.sdk.client.api.config.OpcUaClientConfigBuilder;
+import org.eclipse.milo.opcua.stack.client.DiscoveryClient;
+import org.eclipse.milo.opcua.stack.core.UaException;
+import org.eclipse.milo.opcua.stack.core.types.builtin.DataValue;
+import org.eclipse.milo.opcua.stack.core.types.builtin.NodeId;
+import org.eclipse.milo.opcua.stack.core.types.builtin.Variant;
+import org.eclipse.milo.opcua.stack.core.types.structured.EndpointDescription;
+import org.eclipse.milo.opcua.stack.core.util.EndpointUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PostConstruct;
+import java.util.*;
+import java.util.concurrent.ExecutionException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Service("machine")
+public class MachineServiceImpl implements IMachineService {
+    static Logger logger = Logger.getLogger(MachineServiceImpl.class.getName());
+    @Autowired
+    ConnectionRepository connectionRepository;
+    Map<String, Thread> autobrewers = new HashMap<>();
+    private Connection currentConnection;
+    private OpcUaClient client;
```

Comment:
Why do we have these as fields? The `void` methods should just return objects instead.

Reply:
So that we dont have to create a new OpcUaClient object if the one we currently have is the correct one.

Reply:
You should handle every call as uniqely new, even if it is not.